lib/smart_proxy_reports/ansible_processor.rb

```diff
@@ -71,14 +76,22 @@ def spool_report
       payload = measure :format do
         report_hash.to_json
       end
-      SpooledHttpClient.instance.spool(report_id, payload)
+      SpooledHttpClient.instance.spool("report", report_id, payload)
     end
 
     private
 
     def process_facts(result)
       # TODO: add fact processing and sending to the fact endpoint
-      result["result"]["ansible_facts"] = {}
+      payload = {
+        name: @body["host"],
+        facts: {
+            ansible_facts: result["result"]["ansible_facts"],
+            _type: "ansible",
+            _timestamp: get_now()
+        },
+      }
+      SpooledHttpClient.instance.spool("fact", report_id, payload)
     end
```

Comment:
For easier testing, I suggest to break this down similarly to report parsing into two methods: `build_facts` and `spool_facts`. Then you can use the `build_facts` in tests to easily test facts.